### PR TITLE
Check if database exists before replicate

### PR DESCRIPTION
**Pull request by TheSamSmith**
Fix for #25 

Added check to make sure that the database exists before the replicate. Also changed npm t to beautify and fix the linting as part of it. 

NOTE: I have told Istanbul to ignore the else branch that doesn't exist on the change because I can't see a good way to test it. What would you suggest @redgeoff ?

### CODE REVIEW

File: scripts/cluster.js [hunk 1]
```diff
@@ -81,15 +81,18 @@ Cluster.prototype._replicateRawDB = function (sourceDB, targetDB) {
...
-  self._log('beginning replication of ' + sourceDB + '...');
-  return self._createDBIfMissing(targetDB).then(function () {
-    // Replicate security first so that security is put in place before data is copied over
-    return self._replicateSecurity(sourceDB, targetDB);
-  }).then(function () {
-    return self._replicateRawDB(sourceDB, targetDB);
-  }).then(function () {
-    self._log('finished replicating ' + sourceDB);
-  });
+  /* istanbul ignore else */
```
Comment (redgeoff) [inline scripts/cluster.js]: Sorry, we need 100% test coverage. Please remove this instanbul ignore and write a test

File: test/node-and-browser.js [hunk 2]
```diff
@@ -176,7 +176,6 @@ describe('node and browser', function () {
       return dataShouldEql();
     });
   };
-
```
Comment (redgeoff) [inline test/node-and-browser.js]: There is whitespace modified in this file, please undo this.

File: package.json [hunk 3]
```diff
@@ -37,7 +37,7 @@
     "browser-coverage-report": "istanbul report --dir cache/coverage/browser --root cache/coverage/browser lcov",
     "browser-coverage-check": "istanbul check-coverage --lines 100 --function 100 --statements 100 --branches 100 cache/coverage/browser/coverage.json",
     "browser-coverage-full-test": "npm run browser-coverage-test && npm run browser-coverage-report && npm run browser-coverage-check",
-    "test": "npm run assert-beautified && npm run jshint && npm run node-full-test && npm run browser-coverage-full-test",
```
Comment (redgeoff) [inline package.json]: Please do not change this. I do not want the tests beautifying as beautified code has to be committed intentionally.

Comment (TheSamSmith) [reply package.json]: Now changed back.

File: test/node-and-browser.js (hunk 2 above)
Comment (TheSamSmith) [reply test/node-and-browser.js]: Also reverted.

**Comment by TheSamSmith** [comment]
@redgeoff I have created a test and have made the other requested changes.

File: package.json [hunk 4]
```diff
@@ -37,7 +37,7 @@
...
+    "test": "npm run assert-beatified && npm run jshint && npm run node-full-test && npm run browser-coverage-full-test",
```
Comment (redgeoff) [inline package.json]: Unfortunately, this is still incorrect. (You misspelled beautified)

File: scripts/cluster.js [hunk 5]
```diff
@@ -79,17 +79,25 @@ Cluster.prototype._replicateRawDB = function (sourceDB, targetDB) {
   });
 };
 
+
```
Comment (redgeoff) [inline scripts/cluster.js]: Please remove extra whitespaces like this.

File: scripts/cluster.js [hunk 6]
```diff
@@ -79,17 +79,25 @@ Cluster.prototype._replicateRawDB = function (sourceDB, targetDB) {
...
-    return self._replicateSecurity(sourceDB, targetDB);
-  }).then(function () {
-    return self._replicateRawDB(sourceDB, targetDB);
-  }).then(function () {
-    self._log('finished replicating ' + sourceDB);
-  });
+  return this._sourceSlouch.db.exists(sourceDB)
+    .then(function (value) {
+      if (value === true) {
+        self._log('beginning replication of ' + sourceDB + '...');
```
Comment (redgeoff) [inline scripts/cluster.js]: To make this easier to test I'd recommend creating a new method `Cluster.prototype._createAndReplicateDB()` and move in lines 88-96. Then call `Cluster.prototype._createAndReplicateDB()` in `Cluster.prototype._replicateDB()`. I'll mention the details of the test below.

File: test/node-and-browser.js [hunk 7]
```diff
@@ -290,4 +290,8 @@ describe('node and browser', function () {
     });
   });
 
+  it('should check that the database still exists before calling replicate', function () {
+    return expect(cluster._replicateDB("aaa", "db1"));
```
Comment (redgeoff) [inline test/node-and-browser.js]: I don't think the `expect()` call is doing anything here.

After refactoring per above, you can then create a spy and test:
```js
var createdAndReplicatedDB = false;
cluster._createAndReplicateDB = function () {
  createdAndReplicatedDB = true;
  return;
};

return cluster._replicateDB('aaa', 'db1').then(function () {
  createdAndReplicatedDB.should.eql(true)
});
```

Comment (redgeoff) [reply test/node-and-browser.js]: Thanks for sticking with it!

File: scripts/cluster.js [hunk 8]
```diff
@@ -79,7 +79,7 @@ Cluster.prototype._replicateRawDB = function (sourceDB, targetDB) {
   });
 };
 
-Cluster.prototype._replicateDB = function (sourceDB, targetDB) {
+Cluster.prototype._createdAndReplicateDB = function (sourceDB, targetDB) {
```
Comment (redgeoff) [inline scripts/cluster.js]: Could you please rename this from `_createdAndReplicateDB` to `_createAndReplicateDB` (no d)?

File: test/node-and-browser.js (hunk 2 above)
Comment (redgeoff) [reply test/node-and-browser.js]: It looks like you removed a newline here, please restore this.

**Comment by redgeoff** [comment]
Replaced by https://github.com/redgeoff/replicate-couchdb-cluster/pull/33

@TheSamSmith, thanks!!

**Comment by TheSamSmith** [comment]
Can you please push this to NPM so I can pick up the changes?

**Comment by redgeoff** [comment]
Already did ;)

**Pull request closed without merging**